HabitTracker/Program.cs

```diff
@@ -0,0 +1,133 @@
+            case "d":
+                DeleteEntry();
+                break;
+            case "u":
+                UpdateEntry();
+                break;
+            case "c":
+                CreateHabit();
+                break;
+            case "0":
+                ExitApp();
+                break;
+            default:
+                break;
+        }
+    }
+
+    private static void ViewTable()
+    {
+        List<Entry> entries = dal.GetEntries();
+        string output = string.Empty;
+        foreach (Entry entry in entries)
+        {
+            output += $"{entry}\n";
+        }
+        Console.WriteLine(output);
+
+    }
+
+    private static void ViewHighest()
+    {
+        int habitId = InputValidator.GetHabitId();
+        Entry entry = dal.GetHighestEntryForHabit(habitId);
+        Console.WriteLine($"Below is the most {entry.measurement} you have done in one go:");
+        Console.WriteLine(entry);
+    }
+
+    private static void AddEntry()
+    {
+        int habitId = InputValidator.GetHabitId();
+        string date = InputValidator.GetEntryDate();
+        int quantity = InputValidator.GetEntryQuantity(dal.GetHabit(habitId).measurement);
+        dal.AddEntry(date, quantity, habitId);
+        Console.WriteLine("\nSuccessfully added new entry.");
+
+    }
+
+    private static void DeleteEntry()
+    {
+        ViewTable();
+        int id = InputValidator.GetIdForRemoval();
+        dal.DeleteEntry(id);
+    }
+
+    private static void UpdateEntry()
+    {
+        ViewTable();
+        int id = InputValidator.GetIdForUpdate();
+        int habitId = InputValidator.GetHabitId();
+        int quantity = InputValidator.GetEntryQuantity(dal.GetHabit(habitId).measurement);
+        dal.UpdateEntry(id, quantity);
+    }
+
+    private static void CreateHabit()
+    {
+        string name = InputValidator.GetHabitName();
+        string measurement = InputValidator.GetHabitMeasurement();
+        dal.CreateHabit(name, measurement);
+        Console.WriteLine("Successfully added new habit.");
+    }
+
+    private static void ExitApp()
+    {
+        Environment.Exit(0);
+    }
+
+    private static void DisplayOptionsMenu()
+    {
+        Console.WriteLine("\nChoose an action from the following list:");
+        Console.WriteLine("\tv - View your tracker");
+        Console.WriteLine("\th - View your biggest entry for a habit");
+        Console.WriteLine("\ta - Add a new entry");
+        Console.WriteLine("\td - Delete an entry");
+        Console.WriteLine("\tu - Update an entry");
+        Console.WriteLine("\tc - Create new habit");
+        Console.WriteLine("\t0 - Quit this application");
+        Console.Write("Your option? ");
+    }
+
+    private static void InitializeDatabase()
+    {
+        dal.CreateMainTableIfMissing();
+    }
+
+    private static void DisplayTitle()
+    {
+        Console.WriteLine("Habit Tracker\r");
+        Console.WriteLine("-------------\n");
+    }
+}
```

Comment:
All of these should be separated out into appropriate classes instead of being part of the main program.cs. 
The program.cs should be the initializer of the application, but it shouldn't contain all the code like this, but rather the required calls to other classes and methods. It should be fairly minimal.

Reply:
As a follow up for this: I now realized that the project requirements states everything can be in one class. 
So you can disregard many of my comments about seperating things into seperate classes and files. That will be covered more in future projects. 

But it would be good to come back to this project in the future and do the things I mentioned after you get a few more projects down the line!

Reply:
Hi, yes indeed the project requirements did not get into the separation of concerns, so whilst I am aware of the improvements, at the time in the interests of showing a good line of progression I stuck with a simple execution. Nevertheless, the process of going through it and refactoring is still very useful and forcing me to learn better solutions so I will still update the pull request accordingly. Thank you for all your comments and feedback <3